app/lib/features/news/widgets/news_full_view.dart

```diff
@@ -25,9 +25,21 @@ class _NewsVerticalViewState extends ConsumerState<NewsFullView> {
   @override
   void initState() {
     super.initState();
-    _pageController = PageController(
-      initialPage: widget.initialPageIndex,
-    );
+    _pageController = PageController(initialPage: widget.initialPageIndex);
+    //LISTEN for boost index
+    ref.listenManual(boostGoToPageProvider, fireImmediately: true,
+        (previous, next) {
+      if (next != null) {
+        WidgetsBinding.instance.addPostFrameCallback((_) {
+          _pageController.animateToPage(
+            next,
+            curve: Curves.easeIn,
+            duration: const Duration(milliseconds: 500),
+          );
+          ref.read(boostGoToPageProvider.notifier).state = null;
```

Comment:
okay, that seems like a weird way of triggering actions ... setting to a provider to `0` and resetting it to `0` is not very intuitive for a developer to imagine what this does...

additionally, I am wondering if we can't solve this via either navigation or the `initialPageIndex` that is being used above already ...  Have the doubleTap just set the route again ... which should give it the same initialPageIndex but not as `initState` but instead on the `depdendencyUpdate` ... thus if there is a new `0` (or anything), this should trigger the animation? WDYT?

Reply:
Have the doubleTap just set the route again ... which should give it the same initialPageIndex but not as initState but instead on the depdendencyUpdate ... 

`NewsListPage` is actually not a page but a widget so that will work on doubleTap event?

Is that a good way if I go with static method in `NewsListPage`?